packages/web/src/mutations.rs

```diff
@@ -146,6 +159,11 @@ impl WriteMutations for WebsysDom {
         if self.skip_mutations() {
             return;
         }
+
+        // Invoke cleanup closure BEFORE replacing the element
+        #[cfg(feature = "mounted")]
+        self.invoke_cleanup(id);
```

Comment:
I'm not sure this will catch all removals. If we generate mutations efficiently, only the parent should be replaced when diffing from:
```rust
div {
   button { onmounted }
}
```
To:
```rust
p { "node removed" }
```

The virtual dom knows which nodes are recycled, but it doesn't currently show up in mutations. We could add a new `FreeId` mutation which just drops the reference from the js heap instead of relying on the id being re-used later. That would also allow us to garbage collect dom nodes earlier in general